woocommerce/woocommerce-ios
# [TTPUK] About Tap to Pay screen

Part of: #10870 
Merge after: #10888



## Description


We'll add Tap to Pay on iPhone support for UK WooPayments merchants. This PR adds the About Tap to Pay screen, and a button to show it from the Payments menu on stores/devices which support Tap to Pay.

## Testing instructions


Using a build from Xcode, and a UK-based WooPayments store, on an iPhone XS or newer running iOS 16+:

1. Navigate to Menu > Payments
2. Observe that `About Tap to Pay on iPhone` is visible – tap it
3. Try the `Learn more about card readers` button – observe that it opens an authenticated web view to the correct card reader page for the UK.
4. Close the web view and tap `Learn more` at the bottom (related to Tap to Pay) – observe that it opens a webview to the documentation for Tap to Pay
5. Tap the `Set up Tap to Pay on iPhone` button and run through the set up flow, checking that it works.


Repeat the above test with a US WooPayments store. On step 3, observe that there is no button and the `Important information` section is not shown.

Repeat again with a CA WooPayments store. Observe that the `About Tap to Pay on iPhone` row is not available.

N.B. there's a known issue with the authenticated web views for accounts with 2FA on, particularly prevalent in a simulator at least for A11n accounts, where the Reader is opened after you enter a code – this is pre-existing and out of scope here.

N.B. 2 I've asked Joe for a better colour for the panel in dark mode.

N.B. 3 The list formatting is off, but non-trivial to fix. I've broken it out in #10903 


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/ad61281f-63cd-4d2a-8f59-ec7468bdb256



---
